### PR TITLE
Entity not found

### PR DESCRIPTION
**Pull request by mingushead**
Listens for exceptions thrown from weaver-server.

### CODE REVIEW

File: src/socket.coffee [hunk 1]
```diff
@@ -53,10 +53,15 @@ class Socket
 
   # TODO: Handle errors from server
   emit: (key, body) ->
+
+    isError = (object) ->
```
Comment (malamili) [inline src/socket.coffee]: define this function in top of file in line 4 to save resources (not evaluate this on every emit)

File: src/socket.coffee [hunk 2]
```diff
@@ -53,10 +53,15 @@ class Socket
...
+      return false if not object?
```
Comment (malamili) [inline src/socket.coffee]: just add 'object?' in the second check:

`
object? and Object.keys(object).length is 3 and object.code? and object.payload? and object.message?
`

File: src/socket.coffee [hunk 3]
```diff
@@ -53,10 +53,15 @@ class Socket
...
+      Object.keys(object).length is 3 and object.code? and object.payload? and object.message?
+
     new Promise((resolve, reject) =>
       @io.emit(key, body, (response) ->
-        if response is 0
-          resolve()
+        if isError(response)
+          reject(new Error(response.message))
```
Comment (malamili) [inline src/socket.coffee]: what are we doing with code? Shouldnt we do new Error(response)?

File: src/weaver.coffee [hunk 4]
```diff
@@ -84,30 +84,29 @@ class Weaver
   # Core
   # Loads an entity either from the local repository or from the server
   get: (id, opts) ->
-    # Default options
-    opts = {} if not opts?
-    opts.eagerness = 1 if not opts.eagerness?
+      # Default options
```
Comment (malamili) [inline src/weaver.coffee]: why are these changes? according to .editorconfig in the root file a tab is 2 spaces. is some other editor adding spaces?

File: test/weaver/weaver.test.coffee [hunk 5]
```diff
@@ -121,20 +121,23 @@ describe 'Weaver: addPromise', ->
 describe 'Weaver: Creating an entity', ->
 
   it 'should set type as $ROOT if left empty', ->
-    return
+    root = weaver.add()
```
Comment (malamili) [inline test/weaver/weaver.test.coffee]: very nice!

**Comment by malamili** [comment]
Thnx for testing! 👍

File: src/socket.coffee (hunk 3 above)
Comment (mingushead) [inline src/socket.coffee]: Well, the response looks like this:
response =
     code: 404
     message: 'Entity not found'
     payload: {type:"some-type"..}

Which means the exception as seen from the browser looks like this:

screen shot 2016-09-08 at 11 06 31

rejecting with message property of the response however, means the exception will look like this in the browser:

screen shot 2016-09-08 at 11 10 55

Comment (mingushead) [inline src/socket.coffee]: Perhaps I should change the structure of the response coming from the weaver-server? Let me know what you think..

File: src/socket.coffee [hunk 6]
```diff
@@ -53,10 +57,11 @@ class Socket
 
   # TODO: Handle errors from server
   emit: (key, body) ->
+
     new Promise((resolve, reject) =>
       @io.emit(key, body, (response) ->
-        if response is 0
-          resolve()
+        if isError(response)
+          reject(new Error(response.code + ' - ' + response.message))
```
Comment (malamili) [inline src/socket.coffee]: I think its better to reject with just the response:

reject(response)

**Pull request merged**